crl/storer/storer_test.go

```diff
@@ -0,0 +1,324 @@
+package storer
+
+import (
+	"bytes"
+	"context"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"errors"
+	"io"
+	"io/ioutil"
+	"math/big"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/jmhodges/clock"
+	"google.golang.org/grpc"
+	"google.golang.org/protobuf/types/known/emptypb"
+
+	"github.com/letsencrypt/boulder/crl/crl_x509"
+	cspb "github.com/letsencrypt/boulder/crl/storer/proto"
+	"github.com/letsencrypt/boulder/issuance"
+	blog "github.com/letsencrypt/boulder/log"
+	"github.com/letsencrypt/boulder/metrics"
+	"github.com/letsencrypt/boulder/test"
+)
+
+type fakeUploadCRLServerStream struct {
+	grpc.ServerStream
+	input <-chan *cspb.UploadCRLRequest
+}
+
+func (s *fakeUploadCRLServerStream) Recv() (*cspb.UploadCRLRequest, error) {
+	next, ok := <-s.input
+	if !ok {
+		return nil, io.EOF
+	}
+	return next, nil
+}
+
+func (s *fakeUploadCRLServerStream) SendAndClose(*emptypb.Empty) error {
+	return nil
+}
+
+func (s *fakeUploadCRLServerStream) Context() context.Context {
+	return context.Background()
+}
+
+type fakeS3Putter struct {
+	expectBytes []byte
+}
+
+func (p *fakeS3Putter) PutObject(ctx context.Context, params *s3.PutObjectInput, optFns ...func(*s3.Options)) (*s3.PutObjectOutput, error) {
+	recvBytes, err := ioutil.ReadAll(params.Body)
+	if err != nil {
+		return nil, err
+	}
+	if !bytes.Equal(p.expectBytes, recvBytes) {
+		return nil, errors.New("received bytes did not match expectation")
+	}
+	return &s3.PutObjectOutput{}, nil
+}
+
+type brokenS3Putter struct{}
+
+func (p *brokenS3Putter) PutObject(ctx context.Context, params *s3.PutObjectInput, optFns ...func(*s3.Options)) (*s3.PutObjectOutput, error) {
+	return nil, errors.New("sorry")
+}
+
+func TestUploadCRL(t *testing.T) {
```

Comment:
This big test func should be several smaller test funcs.

Reply:
Done. Created a helper function for setup to avoid proliferating too much duplicate code.